core/src/main/scala/chisel3/Printable.scala

```diff
@@ -46,7 +43,7 @@ import java.util.{
 //   Currently impossible because unpack is called before the name is selected
 //   Could be implemented by adding a new format specifier to Firrtl (eg. %m)
 // TODO Should we provide more functions like map and mkPrintable?
-sealed abstract class Printable {
+sealed abstract class Printable extends BaseSim {
```

Comment:
I don't think it's `Printable` that we want to be annotatable--`Printable` is sort of like a "Hardware String" so it's a little funky to annotate that as opposed to the `Printf` statement itself.

 I think we just want `Printf` and `Stop` to have types that extend `BaseSim` like you did with `assert`, `assume`, and `cover`.

Reply:
+1

Reply:
And another thought is, like what you commented in https://github.com/chipsalliance/chisel3/pull/1968#issuecomment-865336046.
Should we use annotation to represent contents for `Printf` and `Stop`?

Reply:
@sequencer I'm going to not deal with `stop` for now, because it's not strictly necessary for asserts/assumes/covers to be correlated with `stop`s.